BruceDevices/firmware
# Pwnagothi fix for Stack smashing protect failure!

#### Proposed Changes ####

Changed the way to call and alloc memory for the JsonDocument.
Hardcoded default faces/names for pwngrid spam, allowing people to choose from FS

The problem was here: ui.cpp
```
  char stats[25] = "FRND 0 (0)"; // 25 is too small for buffering
  if (friends_run > 0) {
    sprintf(stats, "FRND %d (%d) [%s] %s", friends_run, friends_tot,
            last_friend_name, rssi_bars);
  }
```
changed to String operation

@valentin8709 

#### Types of Changes ####

Structural functionality change

#### Testing ####

Need testing, as I don't have a pwnagotchi to test
Procedure:
* Connect your M5 Device into a Serial Monitor and make sure it is logging
* Wifi> Pwnagotchi
* Turn on a pwnagotchi (or more if available) to see if it makes friends and don't crash
* If it finds a friend and don't crash, hit Ok and choose Pwngrid Spam, try Default
* Look into your pwnagochi to see if it is being spammed
* Hit Esc to exit, and try againg.

#### Linked Issues ####

https://github.com/pr3y/Bruce/issues/265
